hcl2template/types.required_plugins.go

```diff
@@ -99,19 +98,6 @@ func (cfg *PackerConfig) decodeImplicitRequiredPluginsBlock(k ComponentKind, blo
 	// this makes this simple.
 	componentName := block.Labels[0]
 
-	store := map[ComponentKind]packer.BasicStore{
```

Comment:
Why is this being removed? I couldn't understand it bymeself.

Reply:
Ah good catch, this is a carry over from my branch split. I will drop it. 

But originally it was removed to support implicitly required plugins with our removing bundled plugins. But I decided to put it back because without it we will start to automatically install plugins for HCL users not using required plugins. This seems like an expected behavior that we might want to defer until after the legacy mode change is released. 🤔